lib/dialects/abstract/query-generator.js

```diff
@@ -795,6 +795,22 @@ const QueryGenerator = {
             } else if (previousModel.rawAttributes !== undefined && previousModel.rawAttributes[item] && item !== previousModel.rawAttributes[item].field) {
               // convert the item attribute from it's alias
               item = previousModel.rawAttributes[item].field;
+            } else if (
+              item.indexOf('.') !== -1
+              && previousModel.rawAttributes !== undefined
+              && previousModel.rawAttributes[item.split('.')[0]].type instanceof DataTypes.JSON
+            ) {
+              // just quote identifiers for now
+              const identifier = this.quoteIdentifiers(previousModel.name  + '.' + previousModel.rawAttributes[item.split('.')[0]].field);
```

Comment:
`item.split('.')` is called three times in total, that is not good for performance.